kafka-connect-s3/src/main/java/io/confluent/connect/s3/storage/S3Storage.java

```diff
@@ -101,16 +106,45 @@ public ClientConfiguration newClientConfiguration(S3SinkConnectorConfig config)
     clientConfiguration.withUserAgentPrefix(version);
     if (StringUtils.isNotBlank(config.getString(S3_PROXY_URL_CONFIG))) {
       S3ProxyConfig proxyConfig = new S3ProxyConfig(config);
+      RetryPolicy fullJitterPolicy = newRetryPolicy(config);
       clientConfiguration.withProtocol(proxyConfig.protocol())
           .withProxyHost(proxyConfig.host())
           .withProxyPort(proxyConfig.port())
           .withProxyUsername(proxyConfig.user())
-          .withProxyPassword(proxyConfig.pass());
+          .withProxyPassword(proxyConfig.pass())
+          .withRetryPolicy(fullJitterPolicy);
     }
 
     return clientConfiguration;
   }
 
+
+  /**
+   * Creates a retry policy, based on full jitter backoff strategy
+   * and default retry condition.
+   *
+   * @param config the S3 configuration.
+   * @return retry policy
+   * @see com.amazonaws.retry.PredefinedRetryPolicies.SDKDefaultRetryCondition
+   * @see PredefinedBackoffStrategies.FullJitterBackoffStrategy
+   */
+  public RetryPolicy newRetryPolicy(S3SinkConnectorConfig config) {
```

Comment:
Should this be public? Maybe protected so it can be overridden in a subclass?

Reply:
I'd call this `fullJitterRetryPolicy` since it's not a generic `newRetryPolicy` method and as @rhauch said I'd made this `protected`

Reply:
@rhauch There is no need to make it public, but to make unit-testing possible in isolated fashion, I made it public(it's also the same as other similar methods -`io.confluent.connect.s3.storage.S3Storage#newClientConfiguration`, `io.confluent.connect.s3.storage.S3Storage#newS3Client`). Since this method doesn't change the internal state of the `S3Storage` I thought it might be not a big harm.

@kkonstantine sure, renamed.